auth/api/http/policies/transport.go

```diff
@@ -100,7 +101,11 @@ func encodeError(_ context.Context, err error, w http.ResponseWriter) {
 
 	if errorVal, ok := err.(errors.Error); ok {
 		w.Header().Set("Content-Type", contentType)
-		if err := json.NewEncoder(w).Encode(apiutil.ErrorRes{Err: errorVal.Msg()}); err != nil {
+		errorMsg := errorVal.Msg()
+		if errorVal.Err() != nil {
+			errorMsg = fmt.Sprintf("%s : %s", errorVal.Msg(), errorVal.Err().Msg())
+		}
+		if err := json.NewEncoder(w).Encode(apiutil.ErrorRes{Err: errorMsg}); err != nil {
 			w.WriteHeader(http.StatusInternalServerError)
 		}
 	}
```

Comment:
This logic just came to my mind when I saw the code, 

```suggestion
	if errorVal, ok := err.(errors.Error); ok && errorVal.Err() != nil {
		w.Header().Set("Content-Type", contentType)
		if err := json.NewEncoder(w).Encode(apiutil.ErrorRes{Err: errorVal.Error()}); err != nil {
			w.WriteHeader(http.StatusInternalServerError)
		}
	}
```

Reply:
Discussed this over meet with Arvindh.